cmd/find/sizePrimary.go

```diff
@@ -0,0 +1,112 @@
+package cmdfind
+
+import (
+	"fmt"
+	"math"
+	"regexp"
+	"strconv"
+	"strings"
+
+	apitypes "github.com/puppetlabs/wash/api/types"
+)
+
+// Use bytesOf instead of a hash so that we generate a more readable
+// panic message
+func bytesOf(unit byte) uint64 {
+	switch unit {
+	case 'c':
+		return 1
+	case 'k':
+		return 1024
+	case 'M':
+		return 1024 * 1024
+	case 'G':
+		return 1024 * 1024 * 1024
+	case 'T':
+		return 1024 * 1024 * 1024 * 1024
+	case 'P':
+		return 1024 * 1024 * 1024 * 1024 * 1024
+	default:
+		panic(fmt.Sprintf("cmdfind.bytesOf received an unexpected unit %v", unit))
+	}
+}
+
+var sizeValueRegex = regexp.MustCompile(`^(\+|-)?((\d+)|(\d+[ckMGTP]))$`)
+
+// sizePrimary => -size (+|-)?((\d+)|(\d+[ckMGTP]))
+//
+// where c => character (byte), k => kilobyte, M => megabyte, G => gigabyte, T => terabyte, P => petabyte
+//
+// Example inputs:
+//   -size 2   (true if the entry's size in 512-byte blocks, rounded up, is 2)
+//   -size +2  (true if the entry's size in 512-byte blocks, rounded up, is greater than 2)
+//   -size -2  (true if the entry's size in 512-byte blocks, rounded up, is less than 2)
+//   -size +1k (true if the entry's size is greater than 1 kilobyte (1024 bytes))
+//
+//nolint
+var sizePrimary = newAtom([]string{"-size"}, func(tokens []string) (Predicate, []string, error) {
+	tokens = tokens[1:]
+	if len(tokens) == 0 {
+		return nil, nil, fmt.Errorf("-size: requires additional arguments")
+	}
+	v := tokens[0]
+	if !sizeValueRegex.MatchString(v) {
+		return nil, nil, fmt.Errorf("-size: %v: illegal size value", v)
+	}
+
+	cmp := v[0]
+	if cmp == '+' || cmp == '-' {
+		v = strings.TrimPrefix(v, string(cmp))
```

Comment:
So this allows `+++1`? Any reason to do that rather than just a string slice `v[1:]`?

Reply:
It doesn't -- `sizeValueRegex` already validated v. That's a good test-case to add though.

Ah yeah, string slice is simpler. Will update.

Reply:
Did string slicing work?